### PR TITLE
New boids art piece

### PR DESCRIPTION
**Pull request by kriglz**

### CODE REVIEW

File: Gallery/ArtKit iOS/CGFloat+Extension.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        return CGFloat.random * (max - min) + min
+    }
+    
+    /// Returns a random floating point number between 0.0 and 1.0.
+    public static var random: CGFloat {
+        return CGFloat(Float(arc4random()) / 0xFFFFFFFF)
+    }
+    
+    /// Returns a normalized to specified value.
+    public func normalize(to value: CGFloat) -> CGFloat {
```
Comment (wvdk) [inline Gallery/ArtKit iOS/CGFloat+Extension.swift]: The word normalize usual has a very context-specific meaning so what this public extension might do it probably not going to be clear to future users.

File: Gallery/Gallery tvOS/Art Piece Collection View Controllers/ArtPieceCollectionGridViewController.swift.orig [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+/// A subclass of `UIViewController` which contains all art piece grid-like collection view with a vertical scrolling direction.
+class ArtPieceCollectionGridViewController: UIViewController {
+
+    // MARK: - Properties
+
+    /// The object that acts as the delegate of the `ArtPieceCollectionViewControllerDelegate`.
+    weak var delegate: ArtPieceCollectionViewControllerDelegate? = nil
+    
+<<<<<<< HEAD
```
Comment (wvdk) [inline Gallery/Gallery tvOS/Art Piece Collection View Controllers/ArtPieceCollectionGridViewController.swift.orig]: Looks like some merge conflict stuff snuck in here

File: Gallery/Gallery tvOS/Art Piece Collection View Controllers/FeaturedArtPieceCollectionViewController.swift.orig [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import UIKit
+
+/// A subclass of `UIViewController` which contains a featured art piece collection view with a horizontal scrolling direction.
+class FeaturedArtPieceCollectionViewController: UIViewController {
+   
+    // MARK: - Properties
+    
+    /// The object that acts as the delegate of the `ArtPieceCollectionViewControllerDelegate`.
+    weak var delegate: ArtPieceCollectionViewControllerDelegate? = nil
+<<<<<<< HEAD
```
Comment (wvdk) [inline Gallery/Gallery tvOS/Art Piece Collection View Controllers/FeaturedArtPieceCollectionViewController.swift.orig]: Here too

File: Gallery/ArtKit iOS/CGFloat+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+extension CGFloat {
+    
+    /// Returns random number from the range.
+    public static func random(min: CGFloat, max: CGFloat) -> CGFloat {
+        return CGFloat.random * (max - min) + min
+    }
+    
+    /// Returns a random floating point number between 0.0 and 1.0.
+    public static var random: CGFloat {
```
[the remark this answers is not in the document]
Comment (kriglz) [reply Gallery/ArtKit iOS/CGFloat+Extension.swift]: Yeah, but we still use Swift 4.1.

File: Gallery/ArtKit iOS/CGFloat+Extension.swift (hunk 1 above)
Comment (kriglz) [reply Gallery/ArtKit iOS/CGFloat+Extension.swift]: I think it is pretty standard term in math. But I did fix it and made it private.

screen shot 2018-09-13 at 10 53 59 am

**Pull request merged**